src/Text/LLVM/PP.hs

```diff
@@ -1279,6 +1298,32 @@ opt :: Bool -> Fmt Doc
 opt True  = id
 opt False = const empty
 
+-- | Print a ValMd' value as a plain signed integer (Int64) if possible.  If the
+-- ValMd' is not representable as an Int64, defer to ValMd' printing (if
+-- canFallBack is True) or print nothing (for when a ValMd is not a valid
+-- representation).
+
+ppInt64ValMd' :: Bool -> Fmt i -> Fmt (ValMd' i)
+ppInt64ValMd' canFallBack pp = go
```

Comment:
I'm surprised that this function needs to be recursive at all. My general understanding of the code is that if there _is_ an integer literal, it will be represented as a top-level `ValMdValue ... ValInteger`, without being nested underneath some other metadata node. If that is indeed true, then all of the non-`ValMdValue ... ValInteger` cases ought to be able to just call out to `ppMdValue'`, right? (Sadly, I don't think any of our existing `llvm-pretty-bc-parser` test cases involve `DISubrange`s of this level of complexity.)

Reply:
I think this is all verging into the gray area of things, including:
* Principled traversal to extract all possible Int64 representations v.s. typical implementation of the llvm-project C code
* Viability of expressing LLVM code from a newer version in an older syntax

If we consider an AST generated by llvm-pretty-bc-parser, it will always be the direct form, but the AST itself is exported and there's no guarantee a client might submit an AST that allowed an alternate format.  As pointed out previously on this PR, it's possible to express non-sensical things due to the broadness of the `ValMd` inherited from the llvm-project; an anathema from a typical Haskeller's perspective, but a practicality from the C-based llvm-project.

[I'm assuming you meant `ppValMd'` instead of `ppMdValue'`]

Since this is passing all current tests, I'm inclined to leave this for the moment to capture possible concerns.  As we broaden the tests in the future, hopefully we can add more `DISubrange` exercises as you mentioned.